sdk/src/main/java/com/mapbox/maps/renderer/gl/GlUtils.kt

```diff
@@ -0,0 +1,54 @@
+package com.mapbox.maps.renderer.gl
+
+import android.opengl.GLES20
+import android.opengl.Matrix
+import com.mapbox.maps.BuildConfig
+import java.nio.ByteBuffer
+import java.nio.ByteOrder
+import java.nio.FloatBuffer
+
+object GlUtils {
+
+  fun FloatArray.toFloatBuffer(): FloatBuffer = ByteBuffer.allocateDirect(size * 4)
+    .order(ByteOrder.nativeOrder())
+    .asFloatBuffer().also {
+      it.put(this@toFloatBuffer)
+      it.rewind()
+    }
+
+  /**
+   * @param type GLES20.GL_VERTEX_SHADER or GLES20.GL_FRAGMENT_SHADER
+   */
+  fun loadShader(type: Int, shaderCode: String) =
+    GLES20.glCreateShader(type).also { shader ->
+      GLES20.glShaderSource(shader, shaderCode)
+      GLES20.glCompileShader(shader)
+    }
+
+  fun checkError(cmd: String? = null) {
+    if (BuildConfig.DEBUG) {
+      when (val error = GLES20.glGetError()) {
```

Comment:
nit: could be as val errorCode = when()

and after: throw RuntimeException("$cmd -> error in gl: $errorCode")

in order to prevent duplications.

Reply:
With such notation it'll print just int numbers
`public static final int GL_INVALID_ENUM                            = 0x0500;` instead of readable names.

But I've rewritten it a bit to `throw RuntimeException` once and used `when` only to map int to strings